framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -479,6 +479,17 @@
      */
     private Map<String, String> activeOptions;
 
+    /**
+     * Should the analysis assume that side effects to a value can change the type of aliased
+     * references?
+     *
+     * <p>For many type systems, once a local variable's type is refined, side effects to the
+     * variable's value do not change the variable's type annotations. For some type systems, a side
+     * effect to the value could change them; set this field to true.
+     */
+    // Not final so that subclasses can set it.
+    public boolean sideEffectsUnrefineAliases = false;
```

Comment:
Since this field is only used by checkers that use dataflow, it should be in GenericAnnotatedTypeFactory.